connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/BatchExecutableProcessor.java

```diff
@@ -218,4 +219,91 @@ public void deactivateBatch(List<RegisteredExecutable> executables) {
       }
     }
   }
+
+  public CompletableFuture<RegisteredExecutable> restartFromContext(
+      RegisteredExecutable registeredExecutable, ConnectorRetryException retryException) {
+    CompletableFuture<RegisteredExecutable> future = new CompletableFuture<>();
+    LOG.warn(
+        "Inbound connector executable has requested its reactivation in {} {}",
+        retryException.getBackoffDuration().getSeconds(),
+        TimeUnit.SECONDS);
+    if (registeredExecutable instanceof RegisteredExecutable.Cancelled cancelled) {
+      InboundConnectorExecutable<InboundConnectorContext> newExecutable =
+          connectorFactory.getInstance(cancelled.context().getDefinition().type());
+      this.reactivationScheduler.schedule(
+          () ->
+              tryRestart(
+                      newExecutable,
+                      cancelled.context(),
+                      retryException.getBackoffDuration(),
+                      retryException.getRetries())
+                  .thenAccept(future::complete)
```

Comment:
can't we just return the future from `tryRestart`?

Reply:
Because Schedulers returns `ScheduledFuture` which inherit from `Future`. `Future` is really inconvenient to work with and I prefer to work with `CompletableFuture`